packages/block-editor/README.md

```diff
@@ -867,13 +867,14 @@ function MyBlock( { attributes, setAttributes } ) {
 }
 ```
 
-`mode` can be one of three options:
+`mode` can be one of four options:
 
 -   `'disabled'`: Prevents editing the block entirely, i.e. it cannot be
     selected.
 -   `'contentOnly'`: Hides all non-content UI, e.g. auxiliary controls in the
     toolbar, the block movers, block settings.
 -   `'default'`: Allows editing the block as normal.
+-   `'syncedPattern'`: Restricts editing of synced pattern blocks to only child blocks that have attribute connections set.
```

Comment:
I think it'd be good to document where `disabled` isn't working well, and then it makes it easier to consider what options there are.

As `useBlockEditingMode` is a public API, there does need to be some careful consideration of the options added to it. Something like `syncedPattern` might be useful internally as a convenient way to solve this problem, but I can't see how plugins would use it.

Reply:
Yeh, I did explore that, but it seemed like a slightly hacky modification of `disabled`. It is probably easier to put up a separate PR that uses `useBlockEditingMode( 'disabled' )`  than try and explain the differences. I will do that tomorrow. It is worth doing that I think to make it easier to decide which is the best approach.

> I had hoped it would be possible to have the pattern block as disabled and then the selected connected inner blocks as contentOnly, with it all orchestrated by the pattern block.

That is basically what this approach will extend to once we have the attribute connections merged in - currently it is just setting all the children as disabled in order to limit the scope to disabling the inline editing of the synced patterns which can be merged as an independent feature to the partial syncing.

Reply:
I will also do some branches with the connected attributes included as it will be useful to see how the different approaches look with the connected innerBlocks actually wired up - should't take much extra work to do that.

Reply:
> It is probably easier to put up a separate PR that uses useBlockEditingMode( 'disabled' ) than try and explain why it seemed slightly hacky. I will do that tomorrow. It is worth doing that I think to make it easier to decide which is the best approach.

Thanks, that would really help weigh up the trade-offs.